pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -2216,6 +2269,37 @@ def _map_web_search_tool_call(
     )
 
 
+def _map_file_search_tool_call(  # pragma: no cover
+    # File Search Tool mapping - only called from API-dependent response processing paths
+    item: responses.ResponseFileSearchToolCall,
+    provider_name: str,
+) -> tuple[BuiltinToolCallPart, BuiltinToolReturnPart]:
+    args: dict[str, Any] | None = None
+
+    result = {
+        'status': item.status,
+    }
+
+    # The OpenAI SDK has incomplete types for FileSearchToolCall.action
```

Comment:
I don't think that field actually exists.

The type from the SDK looks like this:

```
class ResponseFileSearchToolCall(BaseModel):
    id: str
    """The unique ID of the file search tool call."""

    queries: List[str]
    """The queries used to search for files."""

    status: Literal["in_progress", "searching", "completed", "incomplete", "failed"]
    """The status of the file search tool call.

    One of `in_progress`, `searching`, `incomplete` or `failed`,
    """

    type: Literal["file_search_call"]
    """The type of the file search tool call. Always `file_search_call`."""

    results: Optional[List[Result]] = None
    """The results of the file search tool call."""
```

`queries` and `results` should be stored on the call and return parts.